aws/lambda/hud-query-proxy/lambda_function.py

```diff
@@ -0,0 +1,238 @@
+"""
+Query the database backing metrics.pytorch.org for recent commits and statuses,
+then push these to S3 at ossci-job-status/single/{branch}.tar.gz
+
+This fetches the 100 most recent commits for the branch (GitHub doesn't send
+push events for stacks of commits so we'd need to recover those manually), then
+gets their GitHub Actions job statuses and status events (CircleCI, CodeCov,
+Jenkins, etc.)
+
+Use environment variables 'num_commits' and 'branches' to configure the lambda
+"""
+import os
+import re
+import sys
+import gzip
+import json
+import io
+import boto3
+
+from typing import *
+
+import asyncio
+import aiomysql
+
+
+def eprint(*args):
+    print(*args, file=sys.stderr)
+
+
+async def run_query(query: str, params: List[str] = None) -> Dict[str, Any]:
+    eprint(f"Executing '{query}' with params: {params}")
+
+    my_con = await aiomysql.connect(
+        host=os.environ["db_host"],
+        port=3306,
+        user=os.environ["db_user"],
+        password=os.environ["db_password"],
+        db="pytorch",
+        loop=asyncio.get_event_loop(),
+    )
+    async with my_con.cursor(aiomysql.DictCursor) as cur:
+        await cur.execute(query, params)
+        return await cur.fetchall()
+
+
+PR_RE = re.compile(r"(.*)\(#([0-9]+)\)$")
+NUM_COMMITS = int(os.getenv("num_commits", 100))
+BRANCHES = os.getenv("branches", "master,release/1.9,nightly").split(",")
+
+
+async def get_commits(branch):
+    results = await run_query(
+        f"""
+        select head_commit_timestamp,
+            head_commit_id,
+            head_commit_author_username,
+            head_commit_message
+        from push_event
+        where ref = %s
+        order by head_commit_timestamp desc
+        limit %s;
+        """,
+        (branch, NUM_COMMITS),
+    )
+    for row in results:
```

Comment:
Hmm, if `run_query` indeed returns `Dict[str, Any]`, then row is `str`, isn't i?